library/std/src/sys_common/thread_info.rs

```diff
@@ -1,47 +1,41 @@
 #![allow(dead_code)] // stack_guard isn't used right now on all platforms
-#![allow(unused_unsafe)] // thread_local with `const {}` triggers this liny
 
-use crate::cell::RefCell;
+use crate::cell::Cell;
 use crate::sys::thread::guard::Guard;
 use crate::thread::Thread;
 
-struct ThreadInfo {
-    stack_guard: Option<Guard>,
-    thread: Thread,
-}
-
-thread_local! { static THREAD_INFO: RefCell<Option<ThreadInfo>> = const { RefCell::new(None) } }
-
-impl ThreadInfo {
-    fn with<R, F>(f: F) -> Option<R>
-    where
-        F: FnOnce(&mut ThreadInfo) -> R,
-    {
-        THREAD_INFO
-            .try_with(move |thread_info| {
-                let mut thread_info = thread_info.borrow_mut();
-                let thread_info = thread_info.get_or_insert_with(|| ThreadInfo {
-                    stack_guard: None,
-                    thread: Thread::new(None),
-                });
-                f(thread_info)
-            })
-            .ok()
-    }
+thread_local! {
+    static THREAD: Cell<Option<Thread>> = const { Cell::new(None) };
```

Comment:
Maybe `OnceCell` makes sense here? That would allow skipping the take and set in `current_thread()` and `stack_guard()`.

Reply:
Totally! Great idea, I'll change it.